### PR TITLE
Create toc in FAQ

### PR DESCRIPTION
**Pull request by anxolin**
Creates automatically an index on top of the FAQ

I didn't include in this PR any styling. I left the default links. @biocom if you want to improve it, you can do a PR on top of mine :)  

image

## Note for the reviewers 
I tried to use `useRef` first, and it was not working, because it doesn't notify you when the content is attached, so I used the suggested approach:

image

https://reactjs.org/docs/hooks-reference.html#useref 

----
EDIT: I updated this PR several times to the latest changes of the release, and by mistake I also brought develop. Sorry that now u will see some changes in a the swap test, to fix travis. It's a small change, just ignore it's even nice that it gets into the version.

### CODE REVIEW

File: src/custom/pages/Faq/index.tsx [hunk 1]
```diff
@@ -1,6 +1,8 @@
-import React from 'react'
+import React, { useCallback, useRef, useState } from 'react'
 import Page, { Content, Title } from 'components/Page'
 import styled from 'styled-components'
+import { Link } from 'react-router-dom'
+import { HashLink } from 'react-router-hash-link'
```
Comment (W3stside) [inline src/custom/pages/Faq/index.tsx:5]: we have this internally already, why add a library?

File: src/custom/pages/Faq/index.tsx [hunk 2]
```diff
@@ -1,6 +1,8 @@
-import React from 'react'
+import React, { useCallback, useRef, useState } from 'react'
```
Comment (W3stside) [inline src/custom/pages/Faq/index.tsx:1]: `useRef` is unused

File: src/custom/pages/Faq/index.tsx [hunk 3]
```diff
@@ -1,6 +1,8 @@
...
 import Page, { Content, Title } from 'components/Page'
 import styled from 'styled-components'
+import { Link } from 'react-router-dom'
```
Comment (W3stside) [inline src/custom/pages/Faq/index.tsx:4]: Link is unused

**Comment by biocom** [comment]
@anxolin Getting a compile error:
> 'useRef' is declared but its value is never read.  TS6133

File: src/custom/pages/Faq/index.tsx (hunk 1 above)
Comment (anxolin) [reply src/custom/pages/Faq/index.tsx:4]: As we discussed, this is a pretty simple library and straight fw, over 100K weekly downloads so looks like good tested. This adds on top smooth scroll.

Im working on a PR that will do a refactor, of the two links. Let me do it as an independent PR.

File: src/custom/pages/Faq/index.tsx (hunk 2 above)
Comment (anxolin) [reply src/custom/pages/Faq/index.tsx]: yep, forgot to push some things.

File: src/custom/pages/Faq/index.tsx (hunk 3 above)
Comment (anxolin) [reply src/custom/pages/Faq/index.tsx]: same as above

File: src/custom/pages/Faq/index.tsx (hunk 1 above)
Comment (anxolin) [reply src/custom/pages/Faq/index.tsx:4]: https://github.com/gnosis/gp-swap-ui/pull/507

File: src/custom/pages/Faq/index.tsx [hunk 4]
```diff
@@ -22,21 +23,100 @@ const Wrapper = styled.div`
...
+      tocSections.push({
+        section: lastH2,
+        items
+      })
+      items = []
+    }
+  }
+
+  headingNodes.forEach(entry => {
+    console.log(entry, entry.tagName, entry.textContent)
```
Comment (W3stside) [inline src/custom/pages/Faq/index.tsx:54]: do we want this log

Comment (anxolin) [reply src/custom/pages/Faq/index.tsx:54]: nup! will remove

**Pull request merged**